### PR TITLE
Remove unnecessary sleep after exhausted retries

### PR DESCRIPTION
**Pull request by robbavey**
Also fixes up some issues with logging, where the batch size was calculated
incorrectly.

### CODE REVIEW

File: lib/logstash/outputs/kafka.rb [hunk 1]
```diff
@@ -275,13 +275,16 @@ def retrying_send(batch)
       break if failures.empty?
 
       # Otherwise, retry with any failed transmissions
-      batch = failures
-      delay = @retry_backoff_ms / 1000.0
-      logger.info("Sending batch to Kafka failed. Will retry after a delay.", :batch_size => batch.size,
-                  :failures => failures.size, :sleep => delay);
-      sleep(delay)
+      if remaining != nil && remaining < 0
```
Comment (colinsurprenant) [inline lib/logstash/outputs/kafka.rb]: it seems that this condition is not useful here, IMO if we write the condition as:
```
if remaining.nil? || remaining >= 0
  delay = @retry_backoff_ms / 1000.0
  logger.info("Sending batch to Kafka failed. Will retry after a delay.", :batch_size => batch.size, :failures => failures.size, :sleep => delay)
  batch = failures
  sleep(delay)
end
```
and let the initial condition at the beginning of the method would handle the failure case?

Comment (robbavey) [reply lib/logstash/outputs/kafka.rb]: 👍

File: lib/logstash/outputs/kafka.rb [hunk 2]
```diff
@@ -275,13 +275,14 @@ def retrying_send(batch)
...
-    end
-
+      if remaining.nil? || remaining >= 0
+        delay = @retry_backoff_ms / 1000.0
+        logger.info("Sending batch to Kafka failed. Will retry after a delay.", :batch_size => batch.size,
+                                                                                :failures => failures.size,
+                                                                                :sleep => delay)
+        batch = failures
+        sleep(delay)
+      end    end
```
Comment (colinsurprenant) [inline lib/logstash/outputs/kafka.rb]: oups - minor indentation quirk

Comment (robbavey) [reply lib/logstash/outputs/kafka.rb]: 🤦‍♂️

**Pull request closed without merging**